web/packages/teleport/src/components/AuthnDialog/AuthnDialog.tsx

```diff
@@ -33,11 +33,16 @@ export type Props = {
 };
 
 export default function AuthnDialog({
-  mfaState: { options, challenge, submit, attempt, resetAttempt },
+  mfaState: { options, challenge, submit, attempt, cancelAttempt },
   replaceErrorText,
-  onClose,
+  onClose = () => {},
 }: Props) {
-  if (!challenge && attempt.status !== 'error') return;
+  const showMfaDialog =
+    !!challenge ||
```

Comment:
i just want to make sure this is `show mfa dialog if there IS a challenge`? i think that `not` is throwing me off (brain farting)

Reply:
Yeah I don't know why I thought it was more readable this way, updated to:
```
  const showError =
    attempt.status === 'error' && !(attempt.error instanceof MfaCanceledError);

  if (!challenge && !showError) return;
```